aten/src/ATen/native/cuda/ReduceSumProdKernel.cu

```diff
@@ -19,6 +19,33 @@ struct sum_functor {
   }
 };
 
+// jiterated specialization for `complex<Half>`
+const char sum_name[] = "sum";
+template <>
+struct sum_functor<c10::complex<at::Half>> {
+#if AT_USE_JITERATOR()
+  void operator()(TensorIterator& iter) {
+    using scalar_t = c10::complex<at::Half>;
+    std::string func = jiterator_stringify(
+    arg_t combine(arg_t a, arg_t b) {
+      return a + b;
+    }
+    );
+    jitted_gpu_reduce_kernel<sum_name, scalar_t, scalar_t>(
+        iter, func, 0.);
```

Comment:
Implicit upcasting occurs in jiterated code. (also `jitted_gpu_reduce_kernel` doesn't take `accumulation type`.)